src/touchstone/databases/elasticsearch.py

```diff
@@ -1,183 +1,134 @@
-                                                       str(uuid)})
-        for key, value in filters.items():
-            s = s.filter("term", **{str(key): str(value)})
-        if 'exclude' in search_map:
-            for key, value in search_map['exclude'].items():
-                s = s.exclude('match', **{key: value})
-        _logger.debug("Building query")
-        _first_bucket = str(buckets[0].split('.')[0])
-        self._bucket_list.append(_first_bucket)
-        x = s.aggs.bucket(_first_bucket, 'terms', field=str(buckets[0]))
-        _logger.debug("Building buckets")
-        for bucket_name in buckets[1:]:
-            _curr_bucket = bucket_name.split('.')[0]
-            self._bucket_list.append(_curr_bucket)
-            x = x.bucket(_curr_bucket, 'terms', field=bucket_name)
-        _logger.debug("Finished adding buckets to query")
-        _logger.debug("Adding aggregations to query")
+        logger.debug("Finished Initializing Elasticsearch object")
+
+    def gen_result_dict(self, reponse, buckets, aggs, uuid):
+        output_dict = {}
+        input_dict = reponse.aggs.__dict__
+        # Remove .keyword from bucket names
+        buckets = [e.split(".")[0] for e in buckets]
+
+        def build_dict(input_dict, output_dict):
+            # Iterate through buckets and check if that bucket exists in input_dict
+            for b in buckets:
+                if b in input_dict:
+                    output_dict[b] = {}
+                    # Iterate through bucket values and recurse with each nested bucket
+                    for bucket in input_dict[b]["buckets"]:
+                        output_dict[b][bucket["key"]] = {}
+                        build_dict(bucket, output_dict[b][bucket["key"]])
+            # This is supposed to run in the last level
+            for agg in aggs:
+                if agg in input_dict and "values" in input_dict[agg]:
+                    for name, value in input_dict[agg]["values"].items():
+                        agg_name = "{}{}".format(name, agg)
+                        output_dict[agg_name] = {}
+                        output_dict[agg_name][uuid] = value
+                elif agg in input_dict:
+                    output_dict[agg] = {}
+                    output_dict[agg][uuid] = input_dict[agg]["value"]
+
+        build_dict(input_dict["_d_"], output_dict)
+        return output_dict
+
+    def emit_compute_dict(self, uuid, compute_map, index, identifier):
+        """
+        Returns the normalized data from the ES query
+        """
+        output_dict = {}
+        buckets = compute_map['buckets']
+        aggregations = compute_map['aggregations']
+        collate = compute_map['collate']
+        filters = compute_map['filter']
+
+        logger.debug("Initializing search object")
+        kw_identifier = identifier + ".keyword"  # append .keyword
+        s = Search(using=self._conn_object, index=str(index)).query("match", **{kw_identifier: uuid})
```

Comment:
We should replace Search with Scan, which is going to change the structure of the response.